STYLE-GUIDE.md

```diff
@@ -0,0 +1,372 @@
+
+    ```javascript
+    // good:
+    let dog = 'dog';
+    let cat = 'cat';
+
+    // good:
+    let he = "can't";
+    let she = "won't";
+
+    // bad:
+    let dog = "dog";
+    let cat = "cat";
+
+    // bad:
+    let he = 'can\'t';
+    let she = 'won\'t';
+    ```
+
+### Objects and arrays
+
+* For lists, put commas at the end of each newline, not at the beginning of each item in a list
+
+    ```javascript
+    // good:
+    let animals = [
+      'ape',
+      'bat',
+      'cat'
+    ];
+
+    // bad:
+    let animals = [
+        'ape'
+      , 'bat'
+      , 'cat'
+    ];
+    ```
+
+## Language constructs
+
+* Do not use `for...in` statements with the intent of iterating over a list of numeric keys. Use a for-with-semicolons statement in stead.
+
+  ```javascript
+  // good:
+  let list = ['a', 'b', 'c']
+  for (let i = 0; i < list.length; i++) {
+    alert(list[i]);
+  }
+
+  // bad:
+  let list = ['a', 'b', 'c']
+  for (let i in list) {
+    alert(list[i]);
+  }
+  ```
+
+* Never omit braces for statement blocks (although they are technically optional).
+    ```javascript
+    // good:
+    for (key in object) {
+      alert(key);
+    }
+
+    // bad:
+    for (key in object)
+      alert(key);
+    ```
+
+* Always use `===` and `!==`, since `==` and `!=` will automatically convert types in ways you're unlikely to expect.
+
+    ```javascript
+    // good:
+
+    // this comparison evaluates to false, because the number zero is not the same as the empty string.
+    if (0 === '') {
+      alert("looks like they're equal");
+    }
+
+    // bad:
+
+    // This comparison evaluates to true, because after type coercion, zero and the empty string are equal.
+    if (0 == '') {
+      alert("looks like they're equal");
+    }
+    ```
+
+* Prefer function expressions wherever possible. Don't use function declarations unless absolutely necessary.
+* Prefer arrow functions to the function keyword.
+
+    ```javascript
+    // good:
+    let go = () -> {...};
+
+    // not preferred:
+    function stop() {...};
+    ```
+
+### Opening or closing too many blocks at once
+
+* The more blocks you open on a single line, the more your reader needs to remember about the context of what they are reading. Try to resolve your blocks early, and refactor. A good rule is to avoid closing more than two blocks on a single line--three in a pinch.
+
+    ```javascript
+    // avoid:
+    _.ajax(url, {success: () -> {
+      // ...
+    }});
+
+    // prefer:
+    _.ajax(url, {
+      success: () -> {
+        // ...
+      }
+    });
+    ```
+
+## Padding and whitespace
+
+* Generally, we don't care where you put extra spaces, provided they are not distracting.
+* You may use it as padding for visual clarity. If you do though, make sure it's balanced on both sides.
+
+    ```javascript
+    // bad:
+    alert( 'I chose to put visual padding around this string' );
```

Comment:
It sounds like we are saying this is ok, and yet we are saying it is bad. Either say no white space padding after/before opening/closing parens, or change the example to say this first one is ok.